DSCResources/cChocoSource/cChocoSource.psm1

```diff
@@ -0,0 +1,160 @@
+function Get-TargetResource
+{
+    [CmdletBinding()] 
+    [OutputType([System.Collections.Hashtable])]
+    param
+    (
+        [parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [System.String]
+        $Name,   
+        [ValidateSet('Present','Absent')]
+        [System.String]
+        $Ensure='Present',
+        [parameter(Mandatory = $false)]
+        [ValidateNotNullOrEmpty()]
+        [UInt32]
+        $Priority,
+        [parameter(Mandatory = $false)]
+        [ValidateNotNullOrEmpty()]
+        [System.String]
+        $Source
+    )
+
+    Write-Verbose "Start Get-TargetResource"
+
+    #Needs to return a hashtable that returns the current
+    #status of the configuration component
+    $Configuration = @{
+        Name = $Name
+        Priority = $Priority
+        Source = $Source
+    }
+
+    return $Configuration
+}
+
+function Set-TargetResource
+{
+    [CmdletBinding()]    
+    param
+    (
+        [parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [System.String]
+        $Name,   
+        [ValidateSet('Present','Absent')]
+        [System.String]
+        $Ensure='Present',
+        [parameter(Mandatory = $false)]
+        [ValidateNotNullOrEmpty()]
+        [UInt32]
+        $Priority,
+        [parameter(Mandatory = $false)]
+        [ValidateNotNullOrEmpty()]
+        [System.String]
+        $Source
+    )
+    Write-Verbose "Start Set-TargetResource"
+
+	if($Ensure -eq "Present")
+	{
+		if($priority -eq $null)
+		{
+			choco sources add -n"$name" -s"$source"
+		}
+		else
+		{
+			choco sources add -n"$name" -s"$source" --priority=$priority
+		}
+	}
+	else
+	{
+		choco sources remove -n"$name"
+	}
+}
+
+function Test-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Boolean])]
+    param
+    (
+        [parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [System.String]
+        $Name,   
+        [ValidateSet('Present','Absent')]
+        [System.String]
+        $Ensure='Present',
+        [parameter(Mandatory = $false)]
+        [ValidateNotNullOrEmpty()]
+        [UInt32]
+        $Priority,
+        [parameter(Mandatory = $false)]
+        [ValidateNotNullOrEmpty()]
+        [System.String]
+        $Source
+    )
+
+    Write-Verbose "Start Test-TargetResource"
+	$exe = (get-command choco).Source
+	$chocofolder = $exe.Substring(0,$exe.LastIndexOf("\"))
+
+	if( $chocofolder.EndsWith("bin") )
+	{
+		$chocofolder = $chocofolder.Substring(0,$chocofolder.LastIndexOf("\"))
+	}
```

Comment:
Completely forgot that $env:ChocolateyInstall gets set. I'll put another commit in that uses that and if it's null then looks for it like this.